alicloud/service_alicloud_ecs.go

```diff
@@ -503,15 +503,17 @@ func (s *EcsService) DescribeKeyPair(id string) (keyPair ecs.KeyPair, err error)
 }
 
 func (s *EcsService) DescribeKeyPairAttachment(id string) (keyPair ecs.KeyPair, err error) {
-	parts, err := ParseResourceId(id, 2)
-	if err != nil {
-		if IsExpectedErrors(err, []string{"InvalidKeyPair.NotFound"}) {
-			err = WrapErrorf(err, NotFoundMsg, AlibabaCloudSdkGoERROR)
-		}
-		return
+	index := strings.LastIndexByte(id, ':')
+	if (index < 0) {
+		err := WrapError(fmt.Errorf("Invalid Resource Id %s.", id))
+		return nil, err
 	}
 	keyPairName := parts[0]
-	keyPair, err = s.DescribeKeyPair(keyPairName)
+	keyPair, err := s.DescribeKeyPair(keyPairName)
+	if IsExpectedErrors(err, []string{"InvalidKeyPair.NotFound"}) {
+		err = WrapErrorf(err, NotFoundMsg, AlibabaCloudSdkGoERROR)
+		return keyPair, err
+	}
```

Comment:
I move the above `InvalidKeyPair.NotFound` to here, because `InvalidKeyPair.NotFound` should be occurred after `DescribeKeyPair`. This may have been a bug before.